src/common/device_helpers.cuh

```diff
@@ -1290,6 +1290,21 @@ void InclusiveScan(InputIteratorT d_in, OutputIteratorT d_out, ScanOpT scan_op,
                                            num_items, nullptr, false)));
 }
 
+template <typename InIt, typename OutIt, typename Predicate>
+void CopyIf(InIt in_first, InIt in_second, OutIt out_first, Predicate pred) {
+  // We loop over batches because thrust::copy_if cant deal with sizes > 2^31
+  // See thrust issue #1302, #6822
+  size_t max_copy_size = std::numeric_limits<int>::max() / 2;
```

Comment:
Not sure what standard current compiler supports, but this can be a 'constexpr', since max is. See example:

https://en.cppreference.com/w/cpp/language/constant_expression

Reply:
I see "XGBoost now adopts the C++14 standard", it should be allowed.

Reply:
@vkuzmin-uber Thanks for the review, I will open another PR for constexpr.  Most of the time the compiler should be able to eliminate these constants automatically so did not pay enough attention to it, thanks for pointing it out.